pkg/elevator/internal/util/util.go

```diff
@@ -76,3 +85,49 @@ func VersionExposeConfigForVersion(
 	}
 	return catalogv1alpha1.VersionExposeConfig{}, false
 }
+
+func FormDefaults(defaultField *runtime.RawExtension) (defaults map[string]interface{}, err error) {
+	if defaultField != nil {
+		defaults = make(map[string]interface{})
+		if err := yaml.Unmarshal(defaultField.Raw, &defaults); err != nil {
+			return nil, fmt.Errorf("defaults yaml marshalling: %w", err)
+		}
+	}
+	return
+}
+
+func BuildProviderObj(tenantObj *unstructured.Unstructured, providerObj *unstructured.Unstructured, scheme *runtime.Scheme, elevateFields []catalogv1alpha1.FieldPath, defaults interface{}) error {
```

Comment:
```suggestion
func BuildProviderObj(tenantObj *unstructured.Unstructured, providerObj *unstructured.Unstructured, scheme *runtime.Scheme, exposedFields []catalogv1alpha1.FieldPath, defaults interface{}) error {
```

`Build` is probably the wrong semantic here, as this function is not creating a new provider object, but it's just overriding the spec of an existing object, while leaving metadata and status alone.

Reply:
I've made some changes, let me know what you think